artifactory/commands/transferfiles/manager.go

```diff
@@ -76,15 +76,15 @@ func (ftm *transferManager) doTransfer(isProducerConsumer bool, transferAction t
 	var pollingError error
 	go func() {
 		defer runWaitGroup.Done()
-		pollingError = pollUploads(ftm.srcUpService, uploadTokensChan, doneChan, errorChannel)
+		pollingError = pollUploads(ftm.srcUpService, uploadTokensChan, doneChan, &errorsChannelMng)
 	}()
 
 	// Transfer action to execute.
 	runWaitGroup.Add(1)
 	var actionErr error
 	go func() {
 		defer runWaitGroup.Done()
-		actionErr = transferAction(pcDetails, uploadTokensChan, delayUploadHelper{shouldDelayFunctions: ftm.delayUploadComparisonFunctions, delayedArtifactsChannel: delayedArtifactsChannel}, errorChannel)
+		actionErr = transferAction(pcDetails, uploadTokensChan, delayUploadHelper{shouldDelayFunctions: ftm.delayUploadComparisonFunctions, delayedArtifactsChannelMng: &delayedArtifactsChannelMng}, &errorsChannelMng)
 		if !isProducerConsumer {
```

Comment:
This "isProducerConsumer" argument should be completely removed from this function. Having an API that accepts an argument, that directly impacts the "how" this API works, instead of the "what" it is supposed to do, means that this isn't a good design for the API.
I suggest you discuss this with @RobiNino and decide on the appropriate refactoring.